cmd/juju/storage/volume.go

```diff
@@ -35,3 +38,65 @@ func NewVolumeSuperCommand() cmd.Command {
 type VolumeCommandBase struct {
 	StorageCommandBase
 }
+
+// VolumeInfo defines the serialization behaviour of the storage volume (currently, disk) information.
+type VolumeInfo struct {
+	Attachments map[string]AttachmentInfo
+}
+
+type AttachmentInfo struct {
+	Storage     string  `yaml:"storage" json:"storage"`
+	Assigned    bool    `yaml:"assigned" json:"assigned"`
+	Machine     string  `yaml:"machine" json:"machine"`
+	Attached    bool    `yaml:"attached" json:"attached"`
+	DeviceName  string  `yaml:"device-name" json:"device-name"`
+	Size        *uint64 `yaml:"size" json:"size"`
+	FileSystem  string  `yaml:"file-system" json:"file-system"`
+	Provisioned bool    `yaml:"provisioned" json:"provisioned"`
+}
+
+func formatVolumeInfo(all []params.StorageVolume) ([]VolumeInfo, error) {
+	result := make([]VolumeInfo, len(all))
+	for i, one := range all {
+		a, err := formatAttachmentInfo(one.Attachments)
+		if err != nil {
+			return nil, errors.Trace(err)
+		}
+		result[i] = VolumeInfo{
+			Attachments: a,
+		}
+	}
+	return result, nil
+}
+
+func formatAttachmentInfo(all []params.VolumeAttachment) (map[string]AttachmentInfo, error) {
+	result := make(map[string]AttachmentInfo)
+	for _, one := range all {
+		// TODO (anastasiamac 2015-01-31) add similar logic for volume tags
+		// when available
+		storageTag, err := names.ParseStorageTag(one.Storage)
+		if err != nil {
+			return nil, errors.Annotate(err, "invalid storage tag")
+		}
+		storageName, err := names.StorageName(storageTag.Id())
```

Comment:
could just ignore the error here since it's impossible
storageName, _ := names.StorageName(...)